src/librbd/api/Group.cc

```diff
@@ -54,12 +54,12 @@ snap_t get_group_snap_id(I* ictx,
 }
 
 int group_snap_list(librados::IoCtx& group_ioctx, const char *group_name,
-		    std::vector<cls::rbd::GroupSnapshot> *cls_snaps)
+                    std::vector<cls::rbd::GroupSnapshot> *cls_snaps,
+                    bool sort = false)
```

Comment:
I think the "try to sort but don't fail if it's not possible" mode discussed in https://github.com/ceph/ceph/pull/58002#discussion_r1680596225 should be implemented in this PR -- otherwise it's just adding a bunch of unused code.  If you don't want to implement all three modes because the "try to sort and fail if it's not possible" mode would grow a user only with mirroring, then rename this parameter to `try_to_sort`, change the semantics accordingly and pass true from `Group::snap_list()`.

The second `fail_if_not_ordered` parameter can be added later when this is refactored to fit in with the mirroring work.